Alliance.Client/Extensions/VOIP/Behaviors/PBVoiceChatHandlerClient.cs

```diff
@@ -1,1175 +1,1065 @@
+				{
+					continue;
+				}
+
+				Vec3 agentPosition;
+				if (iteratedNetworkPeer.ControlledAgent != null)
+				{
+					agentPosition = iteratedNetworkPeer.ControlledAgent.Position;
+				}
+				else
+				{
+					// If the other player doesn't control an agent (admin case), we will use our own position
+					agentPosition = myPosition;
+					agentPosition.z += 1;
+				}
+
+
+				if (AudioHelper.CanTargetHearSound(agentPosition, myPosition))
+				{
+					int peerVoiceDataIndex = GetPlayerVoiceDataIndex(iteratedMissionPeer);
+
+					if (peerVoiceDataIndex == -1)
+					{
+						AddPlayerToVoiceChat(iteratedMissionPeer);
+					}
+				}
+				else
+				{
+					int peerVoiceDataIndex = GetPlayerVoiceDataIndex(iteratedMissionPeer);
+
+					if (peerVoiceDataIndex == -1) { continue; }
+					RemovePlayerFromVoiceChat(peerVoiceDataIndex);
+
+
+					OnPeerVoiceStatusUpdated.Invoke(iteratedMissionPeer, false, true);
+				}
+			}
+
+			if (!Config.Instance.NoFriend) return;
+
+			foreach (Agent target in Mission.Current.Agents)
+			{
+				if (target.MissionPeer != null) continue;
+
+				if (AudioHelper.CanTargetHearSound(myPosition, target.Position))
+				{
+					int agentVoiceDataIndex = _playerVoiceList.FindIndex(x => x.Agent == target);
+
+					if (agentVoiceDataIndex == -1)
+					{
+						AddBotToVoiceChat(target);
+					}
+				}
+				else
+				{
+					int agentVoiceDataIndex = _playerVoiceList.FindIndex(x => x.Agent == target);
+
+					if (agentVoiceDataIndex == -1) { continue; }
+					RemovePlayerFromVoiceChat(agentVoiceDataIndex);
+
+					OnBotVoiceStatusUpdated.Invoke(target, false, true);
+				}
+			}
+		}
+
+		private void AddBotToVoiceChat(Agent target)
+		{
+			_playerVoiceList.Add(new PlayerVoiceData(null, _audioMixer, target));
+		}
+
+		private void GetVoiceData(byte[] buffer, int bufferSize, out int readBytesLength)
+		{
+			DequeueVoiceData(buffer, bufferSize, out readBytesLength);
+		}
+
+		private void AddPlayerToVoiceChat(MissionPeer missionPeer)
+		{
+			_playerVoiceList.Add(new PlayerVoiceData(missionPeer, _audioMixer));
+		}
+
+		private void RemovePlayerFromVoiceChat(int indexInVoiceDataList)
+		{
+			PlayerVoiceData playerVoiceData = _playerVoiceList[indexInVoiceDataList];
+
+			playerVoiceData.ClearVoiceData(_audioMixer);
+
+			_playerVoiceList.RemoveAt(indexInVoiceDataList);
+		}
+
+		private int GetPlayerVoiceDataIndex(MissionPeer missionPeer)
+		{
+			for (int i = 0; i < _playerVoiceList.Count; i++)
+			{
+				if (_playerVoiceList[i].Peer == missionPeer)
+				{
+					return i;
+				}
+			}
+
+			return -1;
+		}
+
+		public override void OnPlayerDisconnectedFromServer(NetworkCommunicator networkPeer)
+		{
+			base.OnPlayerDisconnectedFromServer(networkPeer);
+			MissionPeer missionPeer = GameNetwork.MyPeer?.GetComponent<MissionPeer>();
+			MissionPeer component = networkPeer.GetComponent<MissionPeer>();
+			if (component?.Team == null || missionPeer?.Team == null)
+			{
+				return;
+			}
```

Comment:
I'm not sure to understand the benefit of this code.
You are retrieving some data and you return nothing.

Reply:
Probably remnants of early versions. Removed.